### ISSUE TITLE
extend test coverage in time_advance-tests

### ISSUE THREAD

**Issue by bmcdanie** [project-asgard/asgard#373]
currently, we test only that the wavelet space solution for each problem at a given timestep matches the MATLAB solution for that problem/timestep. we could fairly simply extend these tests to also test that the difference between wavelet space solution and analytic solution (for problems with analytic solution) also matches. this will ensure we have consistent analytic solutions generated at every timestep.

**Comment by hbrunie** [comment]
Are these tests extended to double and single precision ? (are we testing ASGARD_DOUBLE_PRECISION OFF)?

**Comment by bmcdanie** [comment]
yes - that macro is only used to control precision in main, but time advance tests are templated to run in single and double precision.

**Comment by mkstoyanov** [comment]
New API tests use the L-2 norm (not just the wavelet error) as well as point-wise values. New PDEs have much more tests included.

One can always use more tests, but this is OK for now.
